### PR TITLE
Fix rigidbody mocap parsing

### PR DESCRIPTION
**Pull request by jaredpaquet4ufl**
I believe optitrack.cpp was derived from https://github.com/whoenig/NatNetSDKCrossplatform/blob/master/src/PacketClient.cpp this request resolves an incorrect simplification of  line 661 through line 670

This resolves:
#6 
https://github.com/USC-ACTLab/crazyswarm/issues/364
possibly others

Tested on:
Ubuntu 18.04
Motive 2.2.0
NatNetVersion: 14.140.50.0
ServerVersion: 3.1.0.0

### CODE REVIEW

**Comment by whoenig** [comment]
Good catch, but I don't think your fix is correct in all cases. In https://github.com/whoenig/NatNetSDKCrossplatform/blob/master/src/PacketClient.cpp#L661-L670, there is a variable being read `nMarkers`, which is of type int (i.e., 32 bits). ptr itself is of type char, i.e., 8 bits. Thus, dereferencing `ptr` might not lead to the correct result if more than 255 markers are visible. It would be better to rely on `memcpy`, similar to the original code in PacketClient.cpp.

**Comment by jaredpaquet4ufl** [comment]
Thanks for the feedback. I have updated the code with your recommendations.

File: src/optitrack.cpp [hunk 1]
```diff
@@ -354,7 +354,11 @@ namespace libmotioncapture {
         for (int i=0; i < nMarkerSets; i++)
         {
           ptr += strlen(ptr) + 1;
-          ptr += 12;
+          int nMarkers = 0; memcpy(&nMarkers, ptr, 4); ptr += 4;
+          for(int j=0; j < nMarkers; j++)
+          {
+            ptr += 12;
```
Comment (whoenig) [inline src/optitrack.cpp]: I'd suggest to remove this inner for loop and just add "ptr += nMarkers * 12;".

**Review by whoenig — CHANGES_REQUESTED** [review]
Please change the for-loop to a calculation instead.

**Comment by jaredpaquet4ufl** [comment]
Completed requested changes

**Comment by whoenig** [comment]
Perfect - thank you very much!

**Pull request merged**